### DISCUSSION TITLE
New build need some help

### DISCUSSION THREAD

**Discussion by jtaylor54** [nebhead/PiFire#120] [General]
![IMG_20220917_125443](https://user-images.githubusercontent.com/29518432/190868344-6affeec9-8a22-4a2d-b78b-a4fe33f8a83e.jpg)

I'm getting close with my build. I have it working except getting no temp readings. Tried my own pit boss probes and then bought new ones. Replaced my ADS1115 that I had from a previous project with new ones using the link in the docs. Still no joy.

@nebhead I also get an error trying to download logs. It shows a message saying server fault or server busy.

@weberbox I am using the V3.01 board. How do I enable display of the 3rd probe when I get them working.

Also what is the jumper for above the relay board pins?

Awesome project. You guys have done some amazing work. Can't wait to get smokin'

Jim Taylor

**Comment by nebhead** [comment]
Interesting.  So, maybe let's get those logs manually.  Can you SSH into your pi and go to the logs directory.  If you are on the latest, it's likely at `/usr/local/bin/pifire/logs` 

Grab the following logs:

```
control.err.log
control.out.log
webapp.err.log
webapp.out.log
```

Send them over and we'll see what's going on here. 

Thanks!

**Reply by jtaylor54** [reply]
Okay here they are..

[control.err.log](https://github.com/nebhead/PiFire/files/9592630/control.err.log)
[control.out.log](https://github.com/nebhead/PiFire/files/9592631/control.out.log)
[webapp.err.log](https://github.com/nebhead/PiFire/files/9592632/webapp.err.log)
[webapp.out.log](https://github.com/nebhead/PiFire/files/9592633/webapp.out.log)

**Comment by nebhead** [comment]
Hmm... redis db seems to be failing.  Can you type the following in ssh:
```bash
sudo apt install redis-server
```
Then maybe reboot the pi and see if that helps.

**Reply by jtaylor54** [reply]
Okay did that and it said 

redis-server is already the newest version (5:6.0.16-1+deb11u2).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.

I rebooted and here are the logs again
[control.err.log](https://github.com/nebhead/PiFire/files/9592677/control.err.log)
[control.out.log](https://github.com/nebhead/PiFire/files/9592678/control.out.log)
[webapp.err.log](https://github.com/nebhead/PiFire/files/9592679/webapp.err.log)
[webapp.out.log](https://github.com/nebhead/PiFire/files/9592680/webapp.out.log)


BTW tx for the help

**Reply by nebhead** [reply]
OK, it still seems to be struggling with the redis-db for some reason.  Let's try to run the `redis-cli` tool and check what keys are loaded.  In ssh again:

```bash
pi@pifire:~ $ redis-cli
127.0.0.1:6379> keys *
1) "updater:percent"
2) "errors"
3) "control:current"
4) "control:tuning"
5) "updater:output"
6) "updater:status"
7) "control:general"
127.0.0.1:6379> 
```

My output will look different from yours, but I just want to see if anything has been put into the redis database. 

> BTW tx for the help

Not a problem!

**Reply by jtaylor54** [reply]
Here goes..

pi@pifire:~ $ redis-cli
127.0.0.1:6379> keys *
1) "control:general"
2) "errors"
3) "metrics:general"
4) "control:current"
5) "control:history"
127.0.0.1:6379>

**Reply by nebhead** [reply]
While in redis-cli, can you execute:

`get errors`

And let me know what your output is?

**Reply by nebhead** [reply]
Also my be helpful to get the list of python packages / versions that are installed.  Quit from redis-cli (type "quit") and type the following:

```
pip3 list 
```

**Reply by jtaylor54** [reply]
Okay get errors shows

127.0.0.1:6379> get errors
"[]"
127.0.0.1:6379>
pi@pifire:~ $ pip3 list
Package            Version
------------------ ---------
ADS1115            0.2.1
async-timeout      4.0.2
bidict             0.22.0
cbor2              5.4.3
certifi            2020.6.20
chardet            4.0.0
ciso8601           2.2.0
click              8.1.3
colorzero          1.1
Deprecated         1.2.13
distro             1.5.0
dnspython          1.16.0
eventlet           0.30.2
Flask              2.2.2
Flask-Mobility     1.1.0
Flask-QRcode       3.1.0
Flask-SocketIO     5.3.1
gpiozero           1.6.2
greenlet           1.1.3
gunicorn           20.1.0
idna               2.10
importlib-metadata 4.12.0
influxdb-client    1.32.0
itsdangerous       2.1.2
Jinja2             3.1.2
luma.core          2.3.1
luma.lcd           2.9.0
MarkupSafe         2.1.1
numpy              1.19.5
packaging          21.3
picamera2          0.3.3
pidng              4.0.9
piexif             1.1.3
Pillow             9.1.1
pip                20.3.4
pushbullet.py      0.12.0
pyftdi             0.54.0
pyparsing          3.0.9
pyserial           3.5
python-apt         2.2.1
python-dateutil    2.8.2
python-engineio    4.3.4
python-magic       0.4.27
python-prctl       1.7
python-socketio    5.7.1
pyusb              1.2.1
qrcode             7.3.1
reactivex          4.0.4
redis              4.3.4
requests           2.25.1
RPi.GPIO           0.7.0
setuptools         52.0.0
simplejpeg         1.6.4
six                1.16.0
smbus2             0.4.2
spidev             3.5
ssh-import-id      5.10
supervisor         4.2.2
typing-extensions  4.3.0
urllib3            1.26.5
uuid               1.30
v4l2-python3       0.3.1
VL53L0X            1.0.4
websocket-client   1.4.1
Werkzeug           2.2.2
wheel              0.34.2
wrapt              1.14.1
zipp               3.8.1

**Reply by nebhead** [reply]
OK, looks like you have a newer version of the Python redis module than me.  I'm going to try installing it on my side and see what happens.

**Reply by jtaylor54** [reply]
It's just a brand new build. I guess that's what gets automatically installed now

**Reply by nebhead** [reply]
Yeah, works fine for me, so it's probably not that.  Hmm... 

In `redis-cli`, let's try looking at `control:general`:

```
pi@pifire:~ $ redis-cli
127.0.0.1:6379> keys *
1) "control:general"
2) "errors"
3) "control:current"
127.0.0.1:6379> get control:general
"{\"updated\": false, \"mode\": \"Stop\", \"s_plus\": true, \"pwm_control\": false, \"duty_cycle\": 100, \"hopper_check\": false, \"recipe\": \"\", \"status\": \"\", \"probe_profile_update\": false, \"settings_update\": false, \"distance_update\": false, \"units_change\": false, \"tuning_mode\": false, \"safety\": {\"startuptemp\": 0, \"afterstarttemp\": 0, \"reigniteretries\": 1, \"reignitelaststate\": \"Smoke\"}, \"setpoints\": {\"grill\": 0, \"probe1\": 0, \"probe2\": 0, \"grill_notify\": 0}, \"notify_req\": {\"grill\": false, \"probe1\": false, \"probe2\": false, \"timer\": false}, \"notify_data\": {\"hopper_low\": false, \"p1_shutdown\": false, \"p2_shutdown\": false, \"timer_shutdown\": false, \"p1_keep_warm\": false, \"p2_keep_warm\": false, \"timer_keep_warm\": false}, \"timer\": {\"start\": 0, \"paused\": 0, \"end\": 0, \"shutdown\": false}, \"manual\": {\"change\": false, \"fan\": false, \"auger\": false, \"igniter\": false, \"power\": false, \"pwm\": 100}, \"errors\": [], \"smartstart\": {\"startuptemp\": 0, \"profile_selected\": 0}}"
```
While in startup or monitor modes, maybe just check:

```
127.0.0.1:6379> hget control:current GrillTemp
"77"
```

**Reply by nebhead** [reply]
Also, just to confirm, you're just on the main branch - not the development branch right?

**Reply by jtaylor54** [reply]
Okay I'll get on it in the am. Just got company and they want to drink a beer so I have to oblige. 🙂

**Reply by nebhead** [reply]
I'm doing a fresh install on a spare Pi Zero W.  I'm using the prototype module and it's working OK.  I'm going to test Celsius mode as I notice your screen looked like it was in degrees C.  And that works fine as well.  

I may need to test it with the actual ADS1115, but that may need to wait until later.  I actually am in the middle of cooking dinner now.  

One other thing we can try is to just do a fresh install again.  Something weird might have happened with your SD card during install or something.

**Reply by nebhead** [reply]
> Okay I'll get on it in the am. Just got company and they want to drink a beer so I have to oblige. slightly_smiling_face

Excellent!  Cheers!  I'm one deep already. :)

**Reply by jtaylor54** [reply]
Okay thanks. I just followed the the instructions on the install and did
wget https://raw.githubusercontent.com/nebhead/pifire/main/auto-install/install.sh
$ bash install.sh

I'm willing to do a renistall if that helps.

Jim

**Reply by nebhead** [reply]
> I'm willing to do a renistall if that helps.

That might be our next step.  Maybe we just start by re-flashing the Raspberry Pi Lite 32-bit OS again and then running the install process again.  

Anyway, enjoy the evening.  Will catch up with you tomorrow.

**Reply by jtaylor54** [reply]
Good morning!

Here is what I did today.

Flashed a Brand new install Raspberry OS Lite 32

sudo apt-get update

sudo apt-get dist-upgrade

sudo raspi-config

   -Enabled I2C
   -Locale us EN UTF8
   -Timezone America/New York

curl https://raw.githubusercontent.com/nebhead/pifire/main/auto-install/install.sh | bash

   -Enabled supervisor WebUI

Rebooted and ran Configuration wizard

   -PiFire Standard
      -Trigger Level High
   -ADS1115 Farenheit
   -ILI9341 TFT with 3-Button Input
      -Button Level HIGH Pullups
   -VL53LOX Time Of Flight Sensor

![Screenshot 2022-09-18 091043](https://user-images.githubusercontent.com/29518432/190910176-a6394d7e-e375-4b85-b252-620b5a6c63b5.jpg)


Completed and rebooted

Still no temp display

Logs
[control.err.log](https://github.com/nebhead/PiFire/files/9594106/control.err.log)
[control.out.log](https://github.com/nebhead/PiFire/files/9594107/control.out.log)
[webapp.err.log](https://github.com/nebhead/PiFire/files/9594108/webapp.err.log)
[webapp.out.log](https://github.com/nebhead/PiFire/files/9594109/webapp.out.log)


Still can't Download Logs
![Server Log Error-Screenshot 2022-09-18 092719](https://user-images.githubusercontent.com/29518432/190910425-5cd46ad9-b1e0-4020-a742-c45fb340f1a8.jpg)


I meant to get a console screen capture of all the upgrade, config and piFire install messages but it rebooted at the end of the piFire Install. I can do a better job and install again if that will help.

Jim

**Reply by jtaylor54** [reply]
Also more info:

pi@pifire:~ $ redis-cli
127.0.0.1:6379> keys *
1) "control:current"
2) "control:general"
3) "errors"

get errors
"[]"
127.0.0.1:6379>

pi@pifire:~ $ pip3 list
Package            Version
------------------ ---------
ADS1115            0.2.1
async-timeout      4.0.2
bidict             0.22.0
cbor2              5.4.3
certifi            2020.6.20
chardet            4.0.0
ciso8601           2.2.0
click              8.1.3
colorzero          1.1
Deprecated         1.2.13
distro             1.5.0
dnspython          1.16.0
eventlet           0.30.2
Flask              2.2.2
Flask-Mobility     1.1.0
Flask-QRcode       3.1.0
Flask-SocketIO     5.3.1
gpiozero           1.6.2
greenlet           1.1.3
gunicorn           20.1.0
idna               2.10
importlib-metadata 4.12.0
influxdb-client    1.32.0
itsdangerous       2.1.2
Jinja2             3.1.2
luma.core          2.3.1
luma.lcd           2.9.0
MarkupSafe         2.1.1
numpy              1.19.5
packaging          21.3
picamera2          0.3.3
pidng              4.0.9
piexif             1.1.3
Pillow             9.1.1
pip                20.3.4
pushbullet.py      0.12.0
pyftdi             0.54.0
pyparsing          3.0.9
pyserial           3.5
python-apt         2.2.1
python-dateutil    2.8.2
python-engineio    4.3.4
python-magic       0.4.27
python-prctl       1.7
python-socketio    5.7.1
pyusb              1.2.1
qrcode             7.3.1
reactivex          4.0.4
redis              4.3.4
requests           2.25.1
RPi.GPIO           0.7.0
setuptools         52.0.0
simplejpeg         1.6.4
six                1.16.0
smbus2             0.4.2
spidev             3.5
ssh-import-id      5.10
supervisor         4.2.2
toml               0.10.1
typing-extensions  4.3.0
urllib3            1.26.5
uuid               1.30
v4l2-python3       0.3.1
VL53L0X            1.0.4
websocket-client   1.4.1
Werkzeug           2.2.2
wheel              0.34.2
wrapt              1.14.1
zipp               3.8.1

**Reply by nebhead** [reply]
> Flashed a Brand new install Raspberry OS Lite 32

Excellent!  OK, so based on the new logs, everything looks to be running just fine and there are no issues with redis now.  Unfortunately, if you are still not getting temperature out, that is an issue.  

Let's go down the path of checking the actual ADS1115 functionality.  Paste the below into `test.py` and run `python3 test.py` from ssh.  

```python3
import ADS1115
import time

ads = ADS1115.ADS1115()

adc_value = [0,0,0,0]

for index in range(4):
    time.sleep(0.05)
    adc_value[index] = ads.readADCSingleEnded(index)
    print(f'adc[{index}] = {adc_value[index]}')
```

> Still can't Download Logs ![Server Log Error-Screenshot 2022-09-18 092719](https://user-images.githubusercontent.com/29518432/190910425-5cd46ad9-b1e0-4020-a742-c45fb340f1a8.jpg)

Just tried on my side and I failed to download them too.  Appears to be a bug.  I'll fix that up soon.

**Reply by jtaylor54** [reply]
pi@pifire:~ $ python3 test.py
adc[0] = 0.0
adc[1] = 3324.375
adc[2] = 3328.125
adc[3] = 329.4375

**Reply by nebhead** [reply]
> pi@pifire:~ $ python3 test.py adc[0] = 0.0 adc[1] = 3324.375 adc[2] = 3328.125 adc[3] = 329.4375

Thanks - Do you have anything connected to ADC[0]?  Is the grill probe connected?

**Reply by nebhead** [reply]
Let's try a different test script using the adafruit ADS1115 library.  As mentioned in the script below, you will first need to install the adafruit circuitpython library.  So go ahead and do that, then create a file named `2test.py` and paste the contents below into that file.  Then run `python3 2test.py` on your system.  I'm guessing that it's going to give you different output from what you see from the previous script.  

```python3
'''
Pre-req: Install adafruit circuitpython library

	sudo pip3 install adafruit-circuitpython-ads1x15

'''

import time
import board
import busio
import adafruit_ads1x15.ads1015 as ADS
from adafruit_ads1x15.analog_in import AnalogIn

# Create the I2C bus
i2c = busio.I2C(board.SCL, board.SDA)

# Create the ADC object using the I2C bus
ads = ADS.ADS1015(i2c)

channel = [0,0,0,0]
# Create single-ended input on channel 0
channel[0] = AnalogIn(ads, ADS.P0)
channel[1] = AnalogIn(ads, ADS.P1)
channel[2] = AnalogIn(ads, ADS.P2)
channel[3] = AnalogIn(ads, ADS.P3)

# Create differential input between channel 0 and 1
#chan = AnalogIn(ads, ADS.P0, ADS.P1)

for index in range(0, len(channel)):
	print(f'ADC[{index}] = {channel[index].voltage * 1000}')
	time.sleep(0.5)
```

I have a theory that there are some ADS1115's that are now being shipped that are different than the originals, and has somehow stopped being compatible with the original ADS1115 python module.  

Anyway, give that a try and let me know what your output looks like.

**Reply by jtaylor54** [reply]
I am using @weberbox V3.01a HAT board.

I had disconnected the 3rd food probe to check my wiring and the probe for anything wonky on my end. To my suprise when I hooked it back up it gave these results

pi@pifire:~ $ python3 test.py
adc[0] = 0.0
adc[1] = 3324.1875
adc[2] = 0.0
adc[3] = 329.4375

I have these probes:

G_PROBE_1
https://smile.amazon.com/dp/B08RDCR6J9?psc=1&ref=ppx_yo2ov_dt_b_product_details

PROBE_1
PROBE_2
https://smile.amazon.com/dp/B0B2QW4JJ3?psc=1&ref=ppx_yo2ov_dt_b_product_details

G-PROBE_2
I have my original PitBoss food probe

As I understood it G_PROBE_1 is the grill temp probe hardwired into the grill and PROBE_1 and PROBE_2 are the flexible food probes.

What is the function of G_PROBE_2 and how is it used?

Jim

**Reply by nebhead** [reply]
> pi@pifire:~ $ python3 test.py adc[0] = 0.0 adc[1] = 3324.1875 adc[2] = 0.0 adc[3] = 329.4375

Interesting results.  So is this right?
 
```
adc[0] = 0.0 => Connected to grill probe (new grill probe listed below) (G_Probe_1)
adc[1] = 3324.1875 => Connected to Food Probe (probe1)
adc[2] = 0.0 => Connected to Food Probe (probe2)
adc[3] = 329.4375 => Connected to PitBoss PT-1000 (G_Probe_2)
```

> I have these probes:
> 
> G_PROBE_1 https://smile.amazon.com/dp/B08RDCR6J9?psc=1&ref=ppx_yo2ov_dt_b_product_details

This should be fine.  I actually like that it's got a quick disconnect in there, but shouldn't be a big deal.  I believe that 
 
> PROBE_1 PROBE_2 https://smile.amazon.com/dp/B0B2QW4JJ3?psc=1&ref=ppx_yo2ov_dt_b_product_details

This may require some tuning, once we get things up and running, but for now we can ignore these.  

> G-PROBE_2 I have my original PitBoss food probe
> 
> As I understood it G_PROBE_1 is the grill temp probe hardwired into the grill and PROBE_1 and PROBE_2 are the flexible food probes.

This is correct.  

> What is the function of G_PROBE_2 and how is it used?

@weberbox added this so that he could use this as a secondary grill probe to either replace G_PROBE_1, or to be averaged with G_PROBE_1.  This is not enabled by default and is ignored.  In the current main version it requires some editing of the settings.json file.  In the development branch, it's selectable in the wizard I think.

**Reply by jtaylor54** [reply]
Okay I had a loose connection and tracing out the ADC connections to the headers realized I did not have the probes really hooked up as I said.

I also changed my probe types all to PT1000 GRILL Probe OEM (Not sure what they should be)


Here is what I have now:
G_PROBE_1 (ADC(0)
https://smile.amazon.com/dp/B08RDCR6J9?psc=1&ref=ppx_yo2ov_dt_b_product_details

PROBE_1 ADC(1)
PROBE_2 ADC(2)
https://smile.amazon.com/dp/B0B2QW4JJ3?psc=1&ref=ppx_yo2ov_dt_b_product_details

G-PROBE_2 ADC(3)
I have my original PitBoss food probe

pi@pifire:~ $ python3 test.py
adc[0] = 332.625
adc[1] = 0.0
adc[2] = 0.0
adc[3] = 0.0

So the grill probe seems to be working okay.

![IMG_20220918_114839](https://user-images.githubusercontent.com/29518432/190916458-c87b5ffd-a77b-4470-8728-4da06277fc26.jpg)

**Reply by nebhead** [reply]
Thanks for the update.  Wiring is definitely finicky with these probes so double checking is always a good idea. 

For the profile settings in PiFire, you'll want the PT1000 OEM probe for the grill probes, as these are wildly different than the food probes.  For a food probe, I'd recommend trying another profile, like the ET-73 profiles - but it's just a guess for those Amazon probes.  You'll likely have to tune those when we get this all setup.  I always recommend going with Thermoworks Pro Series probes.  Yes they are expensive, but they are fantastic and reliable.  

All that said, the little test.py script we made isn't actually using those profiles and is just getting the raw readings from the ADC.  So for anything that is connect or disconnected, I would expect to see some millivolt value on those inputs.  There should be some nominal value on the ADC.  

Try the adafruit script above and let me know how that goes.

**Reply by jtaylor54** [reply]
Okay I have take Granddaughter to get a strep test. Always fun on a Sunday. I'll get back to you later. Thanks for your patience.

**Reply by nebhead** [reply]
> Okay I have take Granddaughter to get a strep test. Always fun on a Sunday. I'll get back to you later. Thanks for your patience.

Was dealing with a sick kid all last week, so I feel the pain.  

I'll be heading out in a bit to go to the local Science Museum with the family, but will be back in the afternoon.

**Reply by jtaylor54** [reply]
Okay I got home and changed my 2 probes to ET73 and rebooted the server.
Then I ran the 2test.py

pi@pifire:~ $ python3 2test.py
ADC[0] = 336.01025421918393
ADC[1] = -2.000061037018952
ADC[2] = -2.000061037018952
ADC[3] = -2.000061037018952

I used my time at the Dr. to order 2 Thermoworks probes. Probably take a week to get.

Jim

**Reply by jtaylor54** [reply]
BTW after reading about other users ADS1115 issues I purchased this last week 

https://smile.amazon.com/dp/B00QIW4MGW?psc=1&ref=ppx_yo2ov_dt_b_product_details

but what they sent was a shorter version with pins on each side and it wouldn't work on the HAT header so I sent it back and got these

https://smile.amazon.com/dp/B082QX1RDV?psc=1&ref=ppx_yo2ov_dt_b_product_details

which seem identical to the one I already had from Aliexpress a couple years old.

Jim

**Reply by nebhead** [reply]
> Okay I got home and changed my 2 probes to ET73 and rebooted the server. Then I ran the 2test.py
> 
> pi@pifire:~ $ python3 2test.py ADC[0] = 336.01025421918393 ADC[1] = -2.000061037018952 ADC[2] = -2.000061037018952 ADC[3] = -2.000061037018952
> 
> I used my time at the Dr. to order 2 Thermoworks probes. Probably take a week to get.
> 
> Jim

Do you have anything plugged into adc 1-3?  Can you run the script multiple times and make sure it's consistent?

**Reply by jtaylor54** [reply]
I unplugged G_Probe2 and reran multiple times

pi@pifire:~ $ python3 2test.py
ADC[0] = 336.01025421918393
ADC[1] = -2.000061037018952
ADC[2] = -2.000061037018952
ADC[3] = 3322.1013824884794
pi@pifire:~ $ python3 2test.py
ADC[0] = 336.01025421918393
ADC[1] = -2.000061037018952
ADC[2] = -2.000061037018952
ADC[3] = 3322.1013824884794
pi@pifire:~ $ python3 2test.py
ADC[0] = 336.01025421918393
ADC[1] = -2.000061037018952
ADC[2] = -2.000061037018952
ADC[3] = 2214.06756797998
pi@pifire:~ $ python3 2test.py
ADC[0] = 336.01025421918393
ADC[1] = -2.000061037018952
ADC[2] = -2.000061037018952
ADC[3] = 3322.1013824884794
pi@pifire:~ $ python3 2test.py
ADC[0] = 336.01025421918393
ADC[1] = -2.000061037018952
ADC[2] = -2.000061037018952
ADC[3] = 3322.1013824884794
pi@pifire:~ $

**Reply by nebhead** [reply]
Well, at least the values are consistent now.  Do you have something plugged into all of those ports?  Or just the ADC0 and ADC3?

**Reply by jtaylor54** [reply]
I had the grill probe in adc(0) and the two Amazon probes (plus I also tried it with my 2 pitboss old probes) plugged into 1 and 2. Nothing on 3. Crazy.

So the grill probe was working well. I shutdown and disconnected the other 2 probes. I hooked it all up to my smoker and fired it up but the grill probe stopped working and reads zero. I did get smoke but shutdown due to under temp. So that's encouraging. Gonna give it a rest for now and watch some football. Put it back on the bench tomorrow and see what works.

**Reply by nebhead** [reply]
> I had the grill probe in adc(0) and the two Amazon probes (plus I also tried it with my 2 pitboss old probes) plugged into 1 and 2. Nothing on 3. Crazy.

Well, ADC3 at 3.3V actually makes sense, since that would be what we would expect with nothing plugged in.  So that's great.  We are making progress.  ADC0 also looks good because it's getting what I would expect for a PT1000 style grill probe.  Similar to my readings as well.  The only strange readings are the probe readings with a -2 value that doesn't make a ton of sense.  If you have a multi-meter, maybe just test the ohms at the plug.  I'd be curious what the resistance is on those.  

> So the grill probe was working well. I shutdown and disconnected the other 2 probes. I hooked it all up to my smoker and fired it up but the grill probe stopped working and reads zero. I did get smoke but shutdown due to under temp. So that's encouraging. Gonna give it a rest for now and watch some football. Put it back on the bench tomorrow and see what works.

Yeah, the ADS1115 python module is failing for some reason.  So we'll need to use the Adafruit module instead.  The good news is that we have a module for this now - and it's currently in the development branch. 

I think what we will need to do is to have you switch to the development branch (in the admin > update screen) and then change the ADC to the version built on Adafruit's python module (admin > config wizard).  This should get you going.  

Once the development branch is ready for prime time - I'll merge it to the main branch so you can go back to that, but it may take a little time.  I think the development branch has been pretty stable for me though.  Regardless, there was one other person have similar problems here and I think that there was a hardware change messing people up.  

Enjoy the remainder of this Sunday.  Hope Monday brings good luck!

**Reply by jtaylor54** [reply]
Okay I tested the probes:
The 2 original Pitboss food probes were 1.083K Ω and 1.088K Ω and ramped up when I applied a flame.
The 2 Amazon food probes were 1.093K Ω and 1.091K Ω and ramped up when I applied a flame.

So I left them disconnected for now and just have G_PROBE_1

I switched to the Dev/Prototype branch
![Screenshot 2022-09-19 093236](https://user-images.githubusercontent.com/29518432/191031318-df16b4df-18e5-4fbd-8a16-9eb4be7c3f69.jpg)

Still no joy. test.py and test2.py fail probably due to the Adafruit ADS1115 module changes I suspect.


[tests.txt](https://github.com/nebhead/PiFire/files/9599402/tests.txt)


Logs
[webapp.out.log](https://github.com/nebhead/PiFire/files/9599410/webapp.out.log)
[control.err.log](https://github.com/nebhead/PiFire/files/9599411/control.err.log)
[control.out.log](https://github.com/nebhead/PiFire/files/9599412/control.out.log)
[webapp.err.log](https://github.com/nebhead/PiFire/files/9599413/webapp.err.log)

**Reply by nebhead** [reply]
OK, so just to make sure, you did the following:

1. Go to the Admin > Updates page and changed the branch:

First check for branches on the remote:
![Screenshot from 2022-09-19 08-20-46](https://user-images.githubusercontent.com/19876886/191053736-dd17dfec-8d12-4773-9508-ae74877ba3e1.png)

Then select the development branch from the dropdown and click checkout branch.
_You may have to reload the page to see the branches show up in the drop down list._ 
![image](https://user-images.githubusercontent.com/19876886/191054306-4a81369d-2b7f-4c99-a320-16bcd7c09ae9.png)

This will go through the update process and restart the server. 
_If you have a first gen Pi Zero W, I sometimes get a Server Error because it's not ready quite yet.  Just give it another 10 seconds or so and reload the page._  

2. Go to the Admin > Config Wizard and reconfigure your system.  

Setup your Platform module to PiFire default, and whatever settings are right for your relays, etc. 
![image](https://user-images.githubusercontent.com/19876886/191055413-b89ba632-917c-4889-b7eb-0720dda3d557.png)

Change to the ADS1x15 module
![Screenshot from 2022-09-19 08-31-42](https://user-images.githubusercontent.com/19876886/191055691-e6c04353-b034-4d56-9565-4fd97d9bb937.png)

Configure the rest of the modules as normal.  

Confirm you have it setup correctly in the final screen (you may have a hopper sensor, I wasn't really sure so I just left mine as prototype/none)
![image](https://user-images.githubusercontent.com/19876886/191056086-3bedda0d-2ede-4322-9aad-ec78735aa9ba.png)

3. Once again, wait for a restart of the server and test again.

**Reply by jtaylor54** [reply]
Okay so I totally screwed that up. Thanks for the pictures. I believe I did it properly this time.

In the event Log I get (plus a red warning box popped up)

2022-09-19 | 12:22:30 | An error occurred configuring the [ads1x15] probes object. The prototype module has been loaded instead. This sometimes means that the hardware is not connected properly, or the module is not configured. Please run the configuration wizard again from the admin panel to fix this issue.

Good news is that the Logs were able to be downloaded so thats fixed.

**Reply by nebhead** [reply]
> Okay so I totally screwed that up. Thanks for the pictures. I believe I did it properly this time.
> 
> In the event Log I get (plus a red warning box popped up)
> 
> 2022-09-19 | 12:22:30 | An error occurred configuring the [ads1x15] probes object. The prototype module has been loaded instead. This sometimes means that the hardware is not connected properly, or the module is not configured. Please run the configuration wizard again from the admin panel to fix this issue.

OK, that is odd that it's not loading anymore.  Can you try to run 2test.py again from ssh?  It should fine, but maybe if we see an error message it will be helpful to track this down. 

> Good news is that the Logs were able to be downloaded so thats fixed.

Excellent. :)  

Thanks for your patience with all of this.  We will get it working.  I am thinking I'm going to order some new ADS1115 modules from amazon to try to see if I can get the same modules that are being shipped these days so that I can do some deeper debug on my end.  I'll probably have to solder up a new PCB, but that's probably something that I would need to do anyway.

**Reply by jtaylor54** [reply]
I sent you $20 to buy the new ADS1115 modules. Least I could do.

**Reply by nebhead** [reply]
Thank you!  You really didn't have to do that, but I greatly appreciate the support.

**Reply by nebhead** [reply]
Just ordered the same ones linked on the hardware documentation page so hopefully that will get me the same ones that you and the other user have.  Meanwhile, I have a spare ADS1115 module that I'm going to try to test with today.

**Reply by nebhead** [reply]
FYI: I soldered up my spare PiFire board, and am testing with another ADS1115 that I have on hand, with the latest PiFire build.  Everything appears to be functioning normally.  I'm expecting that the new ADS1115 parts should arrive tomorrow, so I can see if those behave any differently that my current baseline.  I will keep you posted.  

In the meantime, if you have a chance to let me know what the output looks like when you run the 2test.py currently, with and without the probes connected that would be helpful.  

One thing I noticed on my build was that I had to press down pretty firmly on the ADS1115 module, to ensure it was well seated.  When I initially installed it, there was quite a bit of side-to-side play which obviously can cause some intermittent failures.  Not saying this has anything to do with your challenges, but something to keep in mind.

**Reply by jtaylor54** [reply]
Okay I had to go back to 1.34 main branch as the dev one gave me the ADC11x5 error.

When I run 2test.py I get

[2test_results.txt](https://github.com/nebhead/PiFire/files/9602895/2test_results.txt)

It looks like it can't find the ADC1115 I2C device.

I'm done for today.
Tomorrow I am also soldering up a new board with a better female header for the ADC1115 module. I'm just waiting on them from Amazon. Perhaps it is a loose connection or bad header.

Jim

**Reply by nebhead** [reply]
> It looks like it can't find the ADC1115 I2C device.

Strange - has anything else changed?  Maybe check all the connections and solder joints to see if something is not making good contact. 
 
> I'm done for today. Tomorrow I am also soldering up a new board with a better female header for the ADC1115 module. I'm just waiting on them from Amazon. Perhaps it is a loose connection or bad header.

Sounds good.  I'm hoping my new ADS1115 devices show up at a reasonable hour tomorrow so I can get some testing in before the evening.  Will definitely let you know.

**Reply by nebhead** [reply]
**Update:** Received the new ADS1115 modules just about an hour ago and I've had a bit of time to get it soldered up and plugged into my test rig.  So far, it's working just fine with both the original ADS1115 module and the Adafruit module.  I haven't had much time with it yet, but it yet, but it is behaving normally.  

I'll post some pictures of the new modules later, so you can compare to what you have.  It could be that I got a completely different batch from the Amazon warehouse on the west coast vs. the east coast.  

All that said, it might be a good idea to check for shorts / continuity, etc. on your PCB and through the connectors.  I'm quite busy with the day job for the rest of the day/evening but I will try to check back in.

**Reply by nebhead** [reply]
Some pictures:

Front:
![image](https://user-images.githubusercontent.com/19876886/191375388-d499a311-e8e0-4e2c-871e-01c0ebe4b98d.png)

Back:
![image](https://user-images.githubusercontent.com/19876886/191375493-afa20ef3-0e0c-4681-a09c-6f7b9767181f.png)

**Reply by nebhead** [reply]
I should also mention that I can't really read the markings on the chips without some kind of magnifying glass.  But here is what I think they say: 

OLD - 95 (Texas Instruments Logo) BOGI
NEW - 16 (Texas Instruments Logo) BOGI

**Reply by jtaylor54** [reply]
Yes sorry been quite busy myself. My granddaughter ended up having Covid so dealing with that. I will write up what mine look like and what else I've tried in the AM.

**Reply by nebhead** [reply]
> Yes sorry been quite busy myself. My granddaughter ended up having Covid so dealing with that. I will write up what mine look like and what else I've tried in the AM.

No worries at all!  Take care of the family first and hopefully you can stay healthy too.  Take your time getting back, I'll be here.  Best of luck!

**Reply by nebhead** [reply]
FYI - If you are still running into trouble, I was thinking maybe you could slip one of those ADS1115 modules into an envelope and send it my way.  I can cover the postage.

**Reply by jtaylor54** [reply]
While I have been busy with Covid and also putting up a New Gazebo (with 2000 parts and Chinese Instructions) I did try a few things. I also descended down debug hell for a while where everything you try makes things worse.

I soldered up a new board and also another Pi Zero W (1).  I have an older ADS1115 from EBay 2016 pictured here.
![Ebay-ADS1115-1-25-16](https://user-images.githubusercontent.com/29518432/192143237-d6e9b5c2-516a-4429-b394-948fd2846af7.jpg)

It looks like the new ones (which look like yours) . I can't read any markings even with a magnifying glass.

So after a new build this morning it seems to be working fine. I will hookup the old stuff and see if that also works or I have something wonky.

So I'm good for right now and thanks for all the help and patience.

Jim

**Reply by nebhead** [reply]
> So after a new build this morning it seems to be working fine. I will hookup the old stuff and see if that also works or I have something wonky.

That's great news!  Hope it continues to work well once you get it all buttoned up.  
 
> So I'm good for right now and thanks for all the help and patience.

Excellent.  Just reach out if you need anything else.
